lib/resty/iprepd.lua

```diff
@@ -0,0 +1,78 @@
+local cjson = require('cjson')
+local lrucache = require('resty.lrucache')
```

Comment:
It also mentions that `If the cache hit rate is relatively high, you should use the resty.lrucache class which is faster than resty.lrucache.pureffi`, and I imagine on a lot of services the hit rate will be really high.

Personally, I'm happy to go with either, but I think it may be worth testing and profiling this in the future.

Reply:
On the one hand there should be lots of cache hits for a website, since the client will request both a page and its assets in quick succession. On the other hand there is going to be a huge variation in keys over time as different clients access the site

Possibly it's worth load-testing this module both with all cache hits and all cache misses. Load testing cache hits should be straightforward. To load test all cache misses you could have a client generate random ip addresses and send them in a header like X-Fake-IP, then have nginx call `client:check($http_x_fake_ip)`

The if performance for either case looks worse than expected you could try swapping the cache implementation to see how it affects it or profiling to dig deeper.